### PR TITLE
page contact

### PR DESCRIPTION
**Pull request by Mike69lyon**
création de page contact

### CODE REVIEW

**Review by koni0669 — APPROVED** [review]
ok

**Comment by benrcd** [comment]
Pending conflicts

File: client/src/pages/ContactPage.jsx [hunk 1]
```diff
@@ -1,3 +1,190 @@
...
+      },
+    ];
+
+    let allValid = true;
+    fields.forEach(
+      ({ name, value, message, validate, errorMessage, minLength }) => {
+        if (value.trim() === "") {
+          setError(name, message);
+          allValid = false;
+        } else if (validate && !validate(value)) {
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: explicitez la condition : 

validate === ??? && !validate(value) === ???

File: client/src/pages/ContactPage.jsx [hunk 2]
```diff
@@ -1,3 +1,190 @@
...
+          setError(name, errorMessage);
+          allValid = false;
+        } else if (minLength && value.length < minLength) {
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: explicitez la condition : 

minLength === ??? && ...

File: client/src/pages/ContactPage.jsx [hunk 3]
```diff
@@ -1,3 +1,190 @@
...
+        } else {
+          setSuccess(name);
+        }
+      }
+    );
+    return allValid;
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (validateInputs()) {
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: explicitez la condition : 
validateInputs() === ???

File: client/src/pages/ContactPage.jsx [hunk 4]
```diff
@@ -1,3 +1,190 @@
...
+          <input
+            className="input container"
+            type="text"
+            id="firstname"
+            name="firstname"
+            placeholder="Your firstname please"
+            value={formValues.fullName}
+            onChange={handleChange}
+          />
+          {formErrors.fullName && (
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: expliciter la condition : 

formErrors.fullName === ??? && ...

File: client/src/pages/ContactPage.jsx [hunk 5]
```diff
@@ -1,3 +1,190 @@
...
+          <input
+            className="input container"
+            type="text"
+            id="lastname"
+            name="lasttname"
+            placeholder="Your lastname please"
+            value={formValues.fullName}
+            onChange={handleChange}
+          />
+          {formErrors.fullName && (
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: expliciter la condition

File: client/src/pages/ContactPage.jsx [hunk 6]
```diff
@@ -1,3 +1,190 @@
...
+          <input
+            className="input container"
+            type="email"
+            id="email"
+            name="email"
+            placeholder="Your email address"
+            value={formValues.email}
+            onChange={handleChange}
+          />
+          {formErrors.email && <div className="error">{formErrors.email}</div>}
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: expliciter la condition

File: client/src/pages/ContactPage.jsx [hunk 7]
```diff
@@ -1,3 +1,190 @@
...
+          <input
+            className="input container"
+            type="text"
+            id="subject"
+            name="subject"
+            placeholder="Your subject"
+            value={formValues.subject}
+            onChange={handleChange}
+          />
+          {formErrors.subject && (
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: expliciter la condition

File: client/src/pages/ContactPage.jsx [hunk 8]
```diff
@@ -1,3 +1,190 @@
...
+        <div className="input-control">
+          <textarea
+            className="input-text container"
+            id="contactMessage"
+            name="contactMessage"
+            placeholder="Your message (max 300 characters)*"
+            value={formValues.contactMessage}
+            onChange={handleChange}
+          />
+          {formErrors.contactMessage && (
```
Comment (wildmatthieu) [inline client/src/pages/ContactPage.jsx]: expliciter la condition

**Review by wildmatthieu — CHANGES_REQUESTED** [review]

**Review by Mike69lyon — COMMENTED** [review]
ok

**Review by wildmatthieu — APPROVED** [review]

**Pull request merged**